library/elasticbeanstalk_env.py

```diff
@@ -339,8 +342,7 @@ def main():
                                        option_settings=option_setting_tups,
                                        options_to_remove=None)
 
-                wait_for(ebs, app_name, env_name, wait_timeout, health_is_grey)
-                env = wait_for(ebs, app_name, env_name, wait_timeout, status_is_ready)
+                env = wait_for(ebs, app_name, env_name, wait_timeout, lambda environment: status_is_ready(environment) and version_is_updated(version_label, environment))
```

Comment:
Should we just fail in that case? We would have not managed to update the beanstalk to the version the user has asked for.

If I understand correctly, the current behaviour in this scenario (the beanstalk fails to update, and rolls back) is to pass anyway, this PR would change this to wait until wait_timeout seconds have passed, then fail.